velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -687,18 +774,34 @@ class ExpressionFuzzer {
       auto plan = generateExpression(rootType);
       auto rowVector = generateRowVector();
 
+      // Randomize initial result vector data to test for correct null and data
+      // setting in functions.
+      VectorPtr resultVector;
+      if (vectorFuzzer_.coinToss(0.5)) {
+        resultVector = vectorFuzzer_.fuzzFlat(plan->type());
+      }
+
       // If both paths threw compatible exceptions, we add a try() function to
       // the expression's root and execute it again. This time the expression
       // cannot throw.
-      if (!executeExpression(plan, rowVector, true) && FLAGS_retry_with_try) {
+      if (!executeExpression(
+              plan,
+              rowVector,
+              resultVector ? BaseVector::copy(*resultVector) : nullptr,
+              true) &&
+          FLAGS_retry_with_try) {
         LOG(INFO)
             << "Both paths failed with compatible exceptions. Retrying expression using try().";
 
         plan = std::make_shared<core::CallTypedExpr>(
             plan->type(), std::vector<core::TypedExprPtr>{plan}, "try");
 
         // At this point, the function throws if anything goes wrong.
-        executeExpression(plan, rowVector, false);
+        executeExpression(
+            plan,
+            rowVector,
+            resultVector ? BaseVector::copy(*resultVector) : nullptr,
```

Comment:
By this time resultVector may have changed. Looks like before we would create new results vector on each invocation, which doesn't seem right. I wonder if proper way is to make a copy of the result vector before calling executeExpression for the first time, then use that copy here.

Reply:
I think the current implementation does exactly what you described. 
resultVector is never passed in executeExpression(). Only its copies were passed in so it should stay unchanged.